packages/sw-lib/src/lib/error-factory.js

```diff
@@ -0,0 +1,61 @@
+/*
+ Copyright 2016 Google Inc. All Rights Reserved.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+const errors = {
+  'not-in-sw': 'sw-lib must be loaded in your service worker file.',
+  'unsupported-route-type': 'Routes must be either a express style route ' +
+    'string, a Regex to capture request URLs or a Route instance.',
+  'empty-express-string': 'The Express style route string must have some ' +
+    'characters, an empty string is invalid.',
+};
+
+/**
+ * A simple class to make errors and to help with testing.
+ */
+class ErrorFactory {
+  /**
+   * @param {string} name The error name to be generated.
+   * @param {Error} [thrownError] The thrown error that resulted in this
+   * message.
+   * @return {Error} The generated error.
+   */
+  static createError(name, thrownError) {
+    if (!(name in errors)) {
+      throw new Error(`Unable to generate error '${name}'.`);
+    }
+
+    let message = errors[name];
+    let stack = null;
+    if (thrownError) {
+      message += ` [${thrownError.message}]`;
+      stack = thrownError.stack;
+    }
+
+    const generatedError = new Error();
+    generatedError.name = name;
+    generatedError.message = message;
+    generatedError.stack = stack;
+    return generatedError;
+  }
+
+  /**
+   * Returns the object containing all error names and messages.
+   * @return {Object} Error names and messages.
+   */
+  static getAllErrors() {
```

Comment:
I don't see this as being used anywhere, so consider leaving it out.

If it is something you want to keep, the pattern that we're using elsewhere for this sort of thing is to expose constant values as a named export from a `constants` module, e.g. https://github.com/GoogleChrome/sw-helpers/blob/master/packages/sw-broadcast-cache-update/src/lib/constants.js#L31